File: tz_gobierno/tz_gobierno/report/estado_de_flujo_de_efectivo_digecog/estado_de_flujo_de_efectivo_digecog.py
# Copyright (c) 2026, TZCode S.R.L. and contributors
# For license information, please see license.txt
"""Estado de Flujo de Efectivo por método directo — DIGECOG no acepta el indirecto."""

import frappe
from frappe import _

from tz_gobierno.digecog import estados


def execute(filters=None):
	filters = frappe._dict(filters or {})
	if not filters.company or not filters.from_date or not filters.to_date:
		return [], []

	filas, sin_clasificar = estados.flujo_efectivo(
		filters.company, filters.from_date, filters.to_date
	)

	if sin_clasificar:
		codigos = ", ".join(sorted({c for c, _m in sin_clasificar})[:10])
		frappe.msgprint(
			_(
				"Movimientos de caja con contrapartida sin clasificar: {0}. "
				"Se acumularon en «Otros» de operación para no descuadrar el estado; "
				"conviene mapearlos en digecog/mapeo.py."
			).format(codigos),
			title=_("Clasificación incompleta"),
			indicator="orange",
		)

	if not estados.cuadra_flujo_efectivo(filas):
		frappe.msgprint(
			_("El efectivo inicial más la variación no da el efectivo final."),
			title=_("El estado no cuadra"),
			indicator="red",
		)

	cols = [
		{"fieldname": "concepto", "label": _("Concepto"), "fieldtype": "Data", "width": 520},
		{"fieldname": "monto_actual", "label": _("Monto"), "fieldtype": "Currency", "width": 170},
	]
	return cols, filas

File: tz_gobierno/tz_gobierno/report/estado_de_flujo_de_efectivo_digecog/estado_de_flujo_de_efectivo_digecog.js
// Copyright (c) 2026, TZCode S.R.L. and contributors
// For license information, please see license.txt

frappe.query_reports["Estado de Flujo de Efectivo DIGECOG"] = {
	filters: [
		{
			fieldname: "company",
			label: __("Institución"),
			fieldtype: "Link",
			options: "Company",
			reqd: 1,
			default: frappe.defaults.get_user_default("Company"),
		},
		{
			fieldname: "from_date",
			label: __("Desde"),
			fieldtype: "Date",
			reqd: 1,
			default: frappe.datetime.year_start(),
		},
		{
			fieldname: "to_date",
			label: __("Hasta"),
			fieldtype: "Date",
			reqd: 1,
			default: frappe.datetime.get_today(),
		},
	],
	formatter(value, row, column, data, default_formatter) {
		value = default_formatter(value, row, column, data);
		if (data && data.es_total) {
			value = `<b>${value}</b>`;
		}
		return value;
	},
};
